app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyResponse.java

```diff
@@ -77,24 +97,11 @@ public long getSurveyCreatedOn() {
     public void setSurveyCreatedOn(long surveyCreatedOn) {
         this.surveyCreatedOn = surveyCreatedOn;
     }
-    @Override
-    @JsonIgnore
-    @DynamoDBAttribute
-    public String getHealthCode() {
-        return healthCode;
-    }
-    @Override
-    public void setHealthCode(String healthCode) {
-        this.healthCode = healthCode;
-    }
-    @Override
-    @DynamoDBHashKey
-    @JsonIgnore
-    public String getGuid() {
-        return healthCode + ":" + identifier;
-    }
-    public void setGuid(String guid) {
-        // Ignore. Always calculated as above
+    @DynamoDBIgnore
+    public void setSurveyKey(GuidCreatedOnVersionHolder keys) {
+        this.surveyGuid = keys.getGuid();
+        this.surveyCreatedOn = keys.getCreatedOn();
+        this.surveyKey = String.format("%s:%s", keys.getGuid(), Long.toString(keys.getCreatedOn()));
     }
```

Comment:
Should we have a unit test to enforce that setSurveyKey() modifies the surveyGuid and surveyCreatedOn? (For this method and the setSurveyKey(String), if we decide to change it to work like this one.